service/lib/agama/autoyast/profile_fetcher.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2025] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+# :nodoc:
+module Agama
+  module AutoYaST
+    # Retrieves an AutoYaST profile into an Agama one.
+    #
+    # It supports AutoYaST dynamic profiles features: pre-scripts, rules/classes and ERB.
+    #
+    # It generates an AutoYaST profile that can be converted into an Agama configuration using the
+    # {Agama::AutoYaST::Converter} class.
+    class ProfileFetcher
+      # @param profile_url [String] Profile URL
+      def initialize(profile_url)
+        @profile_url = profile_url
+      end
+
+      # Converts the profile into a set of files that Agama can process.
```

Comment:
Perhaps nitpicking, but the description of this class (and of its public methods) doesn't tell me how it collaborates with other classes.
... ah, this `@return` seems to be the key thing I was missing

```suggestion
      # Converts the profile into a set of files that Agama can process.
      # @return [Hash,nil] an evaluated AutoYaST profile
```